scripts/combine/setupCombine.py

```diff
@@ -144,14 +120,52 @@ def setup(args, inputFile, fitvar, xnorm=False):
         datagroups.groups[base_group].addMembers(datagroups.groups[base_group.replace("mu","tau")].members)
         datagroups.deleteGroup(base_group.replace("mu","tau"))
 
+    if args.fitXsec:
+        datagroups.unconstrainedProcesses.append(base_group)
+
     if xnorm:
         datagroups.select_xnorm_groups(base_group)
-
-    if args.unfolding and args.fitXsec:
-        raise ValueError("Options --unfolding and --fitXsec are incompatible. Please choose one or the other")
-    elif args.fitXsec:
-        datagroups.unconstrainedProcesses.append(base_group)
-    elif args.unfolding and not (args.theoryAgnostic and args.poiAsNoi):
+        if not args.unfolding:
+            # creating the xnorm model (e.g. for the theory fit)
+            if wmass:
+                # add gen charge as additional axis
+                datagroups.groups[base_group].add_member_axis("qGen", datagroups.results, 
+                    member_filters={-1: lambda x: x.name.startswith("Wminus"), 1: lambda x: x.name.startswith("Wplus")}, 
+                    hist_filter=lambda x: x.startswith("xnorm"))
+                xnorm_axes = ["qGen", *datagroups.gen_axes]
+            else:
+                xnorm_axes = datagroups.gen_axes[:]
+            datagroups.setGenAxes([a for a in xnorm_axes if a not in fitvar])
+    
+    if args.poiAsNoi:
+        constrainMass = False if args.theoryAgnostic else True
+        poi_axes = datagroups.gen_axes if args.genAxes is None else args.genAxes
+        # remove specified gen axes from set of gen axes in datagroups so that those are integrated over
+        datagroups.setGenAxes([a for a in datagroups.gen_axes if a not in poi_axes])
```

Comment:
When doing --poiAsNoi and --theoryAgnostic "datagroups.setGenAxes" should not be called. I don't remember why I had removed it (there is a warning few lines below), but it was probably because it was doing something weird with the templates. Maybe it was conflicting with the similar axes appearing in some theory systs like QCD scales, although at some point the signal bin names for pois were modified to avoid this conflict.

Reply:
Yeah, this line actually ensures that the axes specified by args.genAxes are removed from the datagroups.gen_axes.